prov/efa/src/rdm/efa_rdm_ep_fiops.c

```diff
@@ -935,12 +991,11 @@ void efa_rdm_ep_set_use_shm_for_tx(struct efa_rdm_ep *ep)
 	if ((ep->user_info->caps & FI_HMEM)
 	    && hmem_ops[FI_HMEM_CUDA].initialized
 	    && !ep->cuda_api_permitted) {
-		ep->use_shm_for_tx = false;
-		return;
+		use_shm = false;
 	}
 
-	ep->use_shm_for_tx = efa_env.enable_shm_transfer;
```

Comment:
If Neuron or Habana Synapse device is successfully initialized, we should set use_shm to false here, and tear it down. We are currently seeing Neuron performance issues as well... and this seems like the perfect place to fix it.

Reply:
It will still work functionally with EFA, but it will be slow.  I am seeing a ~3% application level regression from keeping SHM on.  There is no "ofi nccl plugin" for trainium, so the only way to turn it off is the env variable.